From 3c92f003c7d1a249cd921a74e5b22297e1368014 Mon Sep 17 00:00:00 2001
From: Zhao Zhili <zhilizhao@tencent.com>
Date: Wed, 14 Oct 2026 15:15:07 +0800
Subject: [PATCH] avformat/iamf_parse: reject a zero parameter rate

param_parse() reads parameter_rate from the bitstream and uses it as
the divisor of av_rescale() when bounding the block duration, so a zero
rate hits av_assert2(c > 0).

Signed-off-by: Zhao Zhili <zhilizhao@tencent.com>
(cherry picked from commit 3c92f003c7d1a249cd921a74e5b22297e1368014)
---
 libavformat/iamf_parse.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/libavformat/iamf_parse.c b/libavformat/iamf_parse.c
index b95c47ce8e3..5338004d999 100644
--- a/libavformat/iamf_parse.c
+++ b/libavformat/iamf_parse.c
@@ -635,6 +635,8 @@ static int param_parse(void *s, IAMFContext *c, AVIOContext *pb,
         }
 
     parameter_rate = ffio_read_leb(pb);
+    if (!parameter_rate)
+        return AVERROR_INVALIDDATA;
     mode = avio_r8(pb) >> 7;
 
     if (mode == 0) {
-- 
2.53.0

